starfish/plot/intensity_table.py

```diff
@@ -0,0 +1,182 @@
+    ax.hist(intensities[Features.DISTANCE].values, bins=bins, **kwargs)
+    annotate_axis(
+        ax,
+        title='distance to nearest\ncode distribution',
+        ylabel='number of features',
+        xlabel='distance to nearest code'
+    )
+    return ax
+
+
+def barcode_magnitude_histogram(
+        intensities,
+        ax: Optional[matplotlib.axes.Axes],
+        log=True,
+        bins=100,
+        **kwargs,
+) -> matplotlib.axes.Axes:
+    """Plot the distribution of barcode magnitudes.
+
+    Parameters
+    ----------
+    intensities : IntensityTable
+        IntensityTable containing features to be summarized
+    ax : Optional[matplotlib.axes.Axes]
+        Axis to plot in. If not provided, defaults to current axis.
+    bins :
+        Number of bins. Default 25.
+    log : bool
+        Whether to plot the y-axis of the histogram in log scale.
+    kwargs : Dict
+        additional keyword arguments to pass to matplotlib.pyplot.hist
+
+    Returns
+    -------
+    matplotlib.axes.Axes :
+        The axis containing the plot
+
+    """
+    ax = ax if ax is not None else plt.gca()
+    magnitudes = feature_trace_magnitudes(intensities, norm_order=2)
+
+    ax.hist(magnitudes, log=log, bins=bins, **kwargs)
+    annotate_axis(
+        ax,
+        title='barcode magnitude\ndistribution',
+        ylabel='number of pixels',
+        xlabel='barcode magnitude'
+    )
+
+    return ax
+
+
+# TODO ambrosejcarr: do we want equal sized axes here?
+def compare_copy_number(
+        intensities,
+        other: pd.Series,
+        ax: Optional[matplotlib.axes.Axes]=None,
+        **kwargs,
+) -> matplotlib.axes.Axes:
+    """
+    Plot the copy number of gene expression across all cells against a reference generated by
+    another computational approach. Report the $r^2$ value.
+
+    Parameters
+    ----------
+    intensities : IntensityTable
+        decoded IntensityTable
+    other : pd.Series
+        A pandas series whose index contains gene names and whose values are expression abundances
+    ax : Optional[matplotlib.axes.Axes]
+        The axis in which to plot the figure. Defaults to the current axis if not provided.
+    kwargs : Dict
+        Additional keyword arguments to pass to sns.regplot
+
+    Returns
+    -------
+    matplotlib.axes.Axes :
+        The axis containing the plot.
+
+    """
+
+    ax = ax if ax is not None else plt.gca()
+    targets, counts = np.unique(intensities[Features.TARGET].values, return_counts=True)
+    this = pd.Series(counts, index=targets)
+    tmp = pd.concat([this, other], join='outer', axis=1, sort=True)
+    tmp.columns = ['result', 'comparison']
+
+    # this seaborn function does not accept an axis, but we can make sure it draws to the right
+    # place by setting the current axis
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', FutureWarning)
+        sns.regplot('comparison', 'result', data=tmp, ax=ax, **kwargs)
```

Comment:
This is a nitpick, but is seaborn necessary here? I think this is the only plot using seaborn, so it may be unnecessary to add an additional dependency for just this plot. I think it shortcuts a bunch of work, so it's probably worth it.

Reply:
>  I think it shortcuts a bunch of work, so it's probably worth it.

Indeed it does. I think it's worth a dependency. If we're adding more plots, eventually we'll use things from it.